Description: Correct use of scripts from vtk7-doc 
 In vtk7-doc these scripts are installed uncompressed. 
 In addition the patch changes the vtk doxygen source directory and
 incorporates an patch to link to the VTK doc and replace PNG by SVG,
 Merged to one patch because they overlap partially. 
Author: Gert Wollny <gw.fossdev@gmail.com>, Mathieu Malaterre <malat@debian.org>

--- a/Utilities/doxygen/vtk/CMakeLists.txt
+++ b/Utilities/doxygen/vtk/CMakeLists.txt
@@ -22,7 +22,7 @@
     set (QT_TAG_FILE "${CMAKE_CURRENT_BINARY_DIR}/Qt.tag")
   endif ()
   set(BUILD_DOXYGEN ON)
-  set(DOCUMENTATION_DOWNLOAD_VTK_TAGFILE ON)
+  set(DOCUMENTATION_DOWNLOAD_VTK_TAGFILE OFF)
   set(DOCUMENTATION_HTML_TARZ ON)
   set(DOXYGEN_PROJECT_NAME "vtkgdcm")
   if(VTK_SOURCE_DIR)
@@ -35,8 +35,8 @@
       # sudo apt-get install vtk-doc
   if(EXISTS "/usr/share/doc/vtk-doc/html")
   set(DOXYGEN_SOURCE_DIR "/usr/share/doc/vtk-doc/html")
-  elseif(EXISTS "/usr/share/doc/vtk6-doc/doxygen")
-  set(DOXYGEN_SOURCE_DIR "/usr/share/doc/vtk6-doc/doxygen")
+  elseif(EXISTS "/usr/share/doc/vtk9/doxygen")
+  set(DOXYGEN_SOURCE_DIR "/usr/share/doc/vtk9/doxygen")
   else()
   message(ERROR "DOXYGEN_SOURCE_DIR need to be set.")
   endif()
@@ -97,7 +97,7 @@
     get_filename_component(DOT_PATH ${DOT} PATH)
   endif()
 
-  set(TAGFILES "\"${CMAKE_CURRENT_BINARY_DIR}/vtkNightlyDoc.tag=http://www.vtk.org/doc/nightly/html\"")
+  #set(TAGFILES "\"${CMAKE_CURRENT_BINARY_DIR}/vtkNightlyDoc.tag=http://www.vtk.org/doc/nightly/html\"")
   if(INCLUDE_QT_DOCUMENTATION)
     set (TAGFILES "${TAGFILES} \"${QT_TAG_FILE}=http://doc.trolltech.com/4.2\"")
   endif()
--- a/Utilities/doxygen/vtk/doc_makeall.sh.in
+++ b/Utilities/doxygen/vtk/doc_makeall.sh.in
@@ -42,7 +42,7 @@
 # Example:
 #   PATH_TO_VTK_DOX_SCRIPTS=@CMAKE_CURRENT_SOURCE_DIR@
 #
-export PATH_TO_VTK_DOX_SCRIPTS="@DOXYGEN_SOURCE_DIR@"
+export PATH_TO_VTK_DOX_SCRIPTS=/usr/share/vtk9/doxygen/
 
 export PATH_TO_VTKGDCM_FINAL_PATH="@VTKGDCM_FINAL_PATH@"
 
@@ -169,12 +169,12 @@
 # Convert the VTK headers to the Doxygen format.
 echo "Convert the VTK headers to the Doxygen format."
 if test "x$PERL_PROG" != "xNOTFOUND" ; then
-    $PERL_PROG "$PATH_TO_VTK_DOX_SCRIPTS/doc_header2doxygen.pl" \
+    $PERL_PROG "$PATH_TO_VTK_DOX_SCRIPTS/doc_header2doxygen.pl"  \
         --to "$INTERMEDIATE_DOX_DIR" \
         --relativeto "@DOXYGEN_SOURCE_LOCATIONS_DIR@" \
         @DOXYGEN_SOURCE_LOCATIONS@ || echoexit "Problem converting VTK headers"
     if test "x@DOXYGEN_BINARY_LOCATIONS_DIR@" != "x"; then
-      $PERL_PROG "$PATH_TO_VTK_DOX_SCRIPTS/doc_header2doxygen.pl" \
+    $PERL_PROG "$PATH_TO_VTK_DOX_SCRIPTS/doc_header2doxygen.pl" \
         --to "$INTERMEDIATE_DOX_DIR" \
         --relativeto "@DOXYGEN_BINARY_LOCATIONS_DIR@" \
         @DOXYGEN_BINARY_LOCATIONS@ || echoexit "Problem converting VTK headers"
@@ -186,13 +186,16 @@
 # Build the full-text index.
 echo "Build the full-text index."
 if test "x$PERL_PROG" != "xNOTFOUND" ; then
+   cp "$PATH_TO_VTK_DOX_SCRIPTS/doc_index.stop" doc_index.stop
+
     $PERL_PROG "$PATH_TO_VTK_DOX_SCRIPTS/doc_index.pl" \
         --project "$PROJECT_NAME" \
-        --stop  "$PATH_TO_VTK_DOX_SCRIPTS/doc_index.stop" \
+        --stop  "doc_index.stop" \
         --store "doc_""$PROJECT_NAME""_index.dox" \
         --to "$INTERMEDIATE_DOX_DIR" \
         @DOXYGEN_SOURCE_LOCATIONS@ \
         @DOXYGEN_BINARY_LOCATIONS@ || echoexit "Problem building full-text index"
+   rm -f doc_index.stop
 fi
 echo "Build the full-text index - done"
 
@@ -201,7 +204,7 @@
 #  # demos that use that class.
 
 #  if test "x$PERL_PROG" != "xNOTFOUND" ; then
-#      $PERL_PROG $PATH_TO_VTK_DOX_SCRIPTS/doc_class2example.pl \
+#      gzip -dc $PATH_TO_VTK_DOX_SCRIPTS/doc_class2example.pl | $PERL_PROG - \
 #          --datamatch "\W((VTK|ParaView)_DATA_ROOT|(VTK|MedVIP)Data|(vtk)?GetDataRoot|ExpandDataFileName)\W" \
 #          --dataicon "paper-clip.gif" \
 #          --dirmatch "^demos$" \
--- a/Utilities/doxygen/vtk/doxyfile.in
+++ b/Utilities/doxygen/vtk/doxyfile.in
@@ -23,6 +23,7 @@
 HAVE_DOT             = YES
 #HAVE_DOT            = NO
 DOT_PATH             = "@DOT_PATH@"
+DOT_IMAGE_FORMAT     = svg
 CLASS_GRAPH          = YES
 COLLABORATION_GRAPH  = YES
 TEMPLATE_RELATIONS   = YES
